aea/helpers/ipfs/base.py

```diff
@@ -43,35 +41,13 @@
     from aea.helpers.ipfs.pb.merkledag_pb2 import PBNode  # type: ignore
 
 
-def _dos2unix(file_content: bytes) -> bytes:
-    """
-    Replace occurrences of Windows line terminator CR/LF with only LF.
-
-    :param file_content: the content of the file.
-    :return: the same content but with the line terminator
-    """
-    return re.sub(b"\r\n", b"\n", file_content, flags=re.M)
-
-
-def _is_text(file_path: str) -> bool:
-    """Check if a file can be read as text or not."""
-    try:
-        with open_file(file_path, "r") as f:
-            f.read()
-        return True
-    except UnicodeDecodeError:
-        return False
-
-
 def _read(file_path: str) -> bytes:
     """Read a file, replacing Windows line endings if it is a text file."""
-    is_text = _is_text(file_path)
     with open(file_path, "rb") as file:
-        file_b = file.read()
-        if is_text:
-            file_b = _dos2unix(file_b)
-
-    return file_b
+        data = file.read()
+    if len(data) == 0:
+        raise ValueError(f"File cannot be empty: {file_path}")
```

Comment:
@angrybayblade why can not be empty?  looks like empty file is a common case.

Reply:
The IPFS daemon treats empty files differently, if you push an empty file to the IPFS node it'll produce different hash compared to using the IPFSHashTool, now the reason behind this unclear I tried looking at the IPFS docs but could not find anything related. So for now we just raise an error when a user tries to hash an empty file so they don't run into these mismatching hashes issue when publishing the packages

Reply:
@angrybayblade ok, let stay as is.
but i see a possible source of problems, if empty file will be needed for some reason